db/compaction_picker_universal.cc

```diff
@@ -568,6 +568,12 @@ Compaction* UniversalCompactionPicker::PickCompactionToReduceSortedRuns(
     output_level = sorted_runs[first_index_after].level - 1;
   }
 
+  // last level is reserved for the files ingested behind
+  if (ioptions_.allow_ingest_behind &&
+      (output_level == vstorage->num_levels() - 1)) {
+    output_level--;
```

Comment:
If you want this only when levels >=3. I think it could be clear to make an assertion before this line. From the code here only we are unable to know this prerequisite.